frontend/public/components/app-contents.tsx

```diff
@@ -87,28 +87,32 @@ const LazyRoute = (props) => (
   />
 );
 
-const getPageRouteExtensions = (perspective: string) =>
-  plugins.registry.getRoutePages().map((r) => {
-    if (r.properties.perspective && r.properties.perspective !== perspective) {
-      return null;
-    }
-    const Component = r.properties.loader ? LazyRoute : Route;
-    return <Component {...r.properties} key={Array.from(r.properties.path).join(',')} />;
-  });
+const getPageRouteExtensions = (perspective: string, flags: FlagsObject) =>
+  plugins.registry
+    .getRoutePages()
+    .filter((e) => plugins.registry.isExtensionInUse(e, flags))
+    .map((r) => {
+      if (r.properties.perspective && r.properties.perspective !== perspective) {
+        return null;
+      }
+      const Component = r.properties.loader ? LazyRoute : Route;
+      return <Component {...r.properties} key={Array.from(r.properties.path).join(',')} />;
+    });
 
 // use `withRouter` to force a re-render when routes change since we are using React.memo
 const AppContents = connect((state: RootState) => ({
   activePerspective: getActivePerspective(state),
+  flags: stateToFlagsObject(state),
 }))(
   withRouter(
-    React.memo(({ activePerspective }) => (
+    React.memo(({ activePerspective, flags }) => (
```

Comment:
I believe this will break `React.memo` because the flags object will be different each time.

Reply:
You're right, `stateToFlagsObject` returns a new object on every call.

On the other hand, the `activePerspective` prop is just a sub-tree of the whole state

```ts
export const getActivePerspective = ({ UI }: RootState): string => UI.get('activePerspective');
```

so I guess that's OK with `React.memo` *but* the resolved TS type of `activePerspective` changes to `never` which is due to untyped `withRouter` call :confused:

Reply:
IIUC React.memo() does a shallow comparison of props and objects of props. So for a flat `flags` object of `{ [FLAG_NAME]: boolean }` it should be fine i think.

https://github.com/facebook/react/blob/v16.8.6/packages/shared/shallowEqual.js

Reply:
Ah, you're right. We should be OK 👍 

> By default it will only shallowly compare complex objects in the props object.

https://reactjs.org/docs/react-api.html#reactmemo